client_integration_test.go

```diff
@@ -873,6 +873,65 @@ func TestClientChmodReadonly(t *testing.T) {
 	}
 }
 
+func TestClientSetuid(t *testing.T) {
+	skipIfWindows(t) // No UNIX permissions.
+	if *testServerImpl {
+		t.Skipf("skipping with -testserver")
+	}
+
+	sftp, cmd := testClient(t, READWRITE, NODELAY)
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	f, err := ioutil.TempFile("", "sftptest-setuid")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+	f.Close()
+
+	const (
+		unixSetuid = 04000
```

Comment:
DRY is not really about avoiding code duplication, it is about avoiding two values/processes that are the same from having inconsistent implementations.

Implementing these two values separately means that there could be inconsistencies between them. This is why I was pushing for the use of the `syscall.S_ISUID` values: we should ideally not be defining these values unless we absolutely have to. (e.g. `S_IFMT` had to be imported locally, because the windows implementation has a bad value. None of _these_ flags are different though, because POSIX has already for a long time established strict values for them.)

Reply:
I repeated the constants here because they should have these exact values, regardless of implementation (via syscall or explicitly). It looks ugly, but it's written this way precisely to catch potential inconsistencies.

Reply:
Again, they are codified in POSIX now, and have been set to the correct values… except `S_IFMT` but we resolved that one ourselves.